src/model/generatorRecipe.ts

```diff
@@ -18,7 +18,7 @@ export class GeneratorRecipe {
     * Length of the generated value
     */
     'length'?: number;
-    'characterSets'?: Set<GeneratorRecipe.CharacterSetsEnum>;
+    'characterSets'?: Array<GeneratorRecipe.CharacterSetsEnum>;
```

Comment:
This solves the immediate serialization problem. We lose the built-in "only include one of each CharacterSet" protection, though.

Could you add a check to the `validRecipe` function in `builder.ts` that makes sure the user is not submitting a recipe like `['SYMBOLS','SYMBOLS']`?

Reply:
Nice observation. Now the `validRecipe` filters the array to have only the unique character sets